examples/flutter_gallery/lib/demo/cupertino/cupertino_picker_demo.dart

```diff
@@ -42,24 +45,90 @@ class _CupertinoPickerDemoState extends State<CupertinoPickerDemo> {
             ),
             child: new Row(
               mainAxisAlignment: MainAxisAlignment.spaceBetween,
-              children: <Widget>[
-                const Text('Favorite Color'),
-                new Text(
-                  coolColorNames[_selectedItemIndex],
-                  style: const TextStyle(color: CupertinoColors.inactiveGray),
-                ),
-              ],
+              children: children,
             ),
           ),
         ),
       ),
     );
   }
 
-  Widget _buildBottomPicker() {
+  Widget _buildColorPicker() {
     final FixedExtentScrollController scrollController =
-        new FixedExtentScrollController(initialItem: _selectedItemIndex);
+      new FixedExtentScrollController(initialItem: _selectedColorIndex);
+    return new CupertinoPicker(
+      scrollController: scrollController,
+      itemExtent: _kPickerItemHeight,
+      backgroundColor: CupertinoColors.white,
+      onSelectedItemChanged: (int index) {
+        setState(() {
+          _selectedColorIndex = index;
+        });
+      },
+      children: new List<Widget>.generate(coolColorNames.length, (int index) {
+        return new Center(child:
+        new Text(coolColorNames[index]),
+        );
+      }),
+    );
+  }
+
+  Widget _buildAlarmPicker() {
+    return new Row(
+      children: <Widget>[
+        new Expanded(
```

Comment:
Removing the wrapping Expanded will invoke the error: BoxConstraints forces an infinite width.
So for simplification I'm wrapping an Expanded around each CupertinoPicker.

Reply:
We should fix this. I'll take a look

Reply:
I see the issue. We can fix this in a future PR since this PR is really close already. 

Design-wise: we may want a multi-column picker in the end but with some of the constructor params extracted out. For instance, it wouldn't make sense (at least as a Cupertino widget) to have side by side pickers that has different magnifications, item extents or height in general etc. It would also be sensible to provide a single onChange callback that snapshots all the columns and a child delegate that deals with multiple columns. 

That widget could use ListWheelScrollView directly and _buildGradientScreen and _buildMagnifierScreen once instead of once per column currently.